pycqed/analysis_v2/base_analysis.py

```diff
@@ -383,7 +404,23 @@ def add_measured_data(raw_data_dict, compression_factor=1):
                 if 'soft_sweep_points' in raw_data_dict:
                     hsl = len(raw_data_dict['hard_sweep_points'])
                     ssl = len(raw_data_dict['soft_sweep_points'])
-                    measured_data = np.reshape(data[i], (ssl, hsl)).T
+                    if hybrid_measurement:
+                        # take out CalibrationPoints from the end of each
+                        # segment, and reshape the remaining data based on the
+                        # hard (1st dimension) and soft (1st dimension)
+                        # sweep points
+                        data_no_cp = data[i][:len(data[i])-len(cp.states)]
+                        measured_data = np.reshape(data_no_cp, (ssl, hsl)).T
+                        if len(cp.states) > 0:
+                            # add back ssl number of copies of the cal points
+                            # at the end of each soft sweep slice
+                            cal_pts = data[i][-len(cp.states):]
+                            cal_pts_arr = np.reshape(np.repeat(cal_pts, ssl),
```

Comment:
Are you sure what you're doing here would work with e.g. active reset and preselection? those readout would not be included in SweepPoints, from which you retrieve the number of readouts for the reshaping, right? Also, as you already corrected somewhere else in the analysis, with preparation params the data of cal_points is not only len(cp.states).

Reply:
Good catch, you are right, it wouldn't....

Reply:
I fixed this now. I don't really have good data to test it on. I have some files from Ants' first tries where the data is just noise basically. The analysis works for these. After that he changed the SweepPoints such that even though he conceptually has a 2D measurement, he only saves one dict in SweepPoints. But my implementation relies on the SweepPoints being 2D.